### PR TITLE
refactor(pubsub): add generated admin subscription samples

### PR DESCRIPTION
**Pull request by alevenberg**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13535)

### CODE REVIEW

**Review by dbolduc — APPROVED** [review]

File: google/cloud/pubsub/samples/subscription_admin_samples.cc [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    command(std::move(client), std::move(argv));
+  };
+  return google::cloud::testing_util::Commands::value_type{name,
+                                                           std::move(adapter)};
+}
+
+void CreateSubscription(
+    google::cloud::pubsub_admin::SubscriptionAdminClient client,
+    std::vector<std::string> const& argv) {
+  //! [create-subscription]
```
Comment (dbolduc) [inline google/cloud/pubsub/samples/subscription_admin_samples.cc:56]: FYI: nothing is referencing these regions.

Comment (alevenberg) [inline google/cloud/pubsub/samples/subscription_admin_samples.cc:56]:
```suggestion
```

File: google/cloud/pubsub/samples/subscription_admin_samples.cc [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    if (sub.status().code() == google::cloud::StatusCode::kAlreadyExists) {
+      std::cout << "The subscription already exists\n";
+      return;
+    }
+    if (!sub) throw std::move(sub).status();
+
+    std::cout << "The subscription was successfully created: "
+              << sub->DebugString() << "\n";
+  }
+  //! [create-subscription]
```
Comment (alevenberg) [inline google/cloud/pubsub/samples/subscription_admin_samples.cc:76]:
```suggestion
```

File: google/cloud/pubsub/samples/subscription_admin_samples.cc [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  (std::move(client), argv.at(0), argv.at(1), argv.at(2));
+}
+
+void DeleteSubscription(
+    google::cloud::pubsub_admin::SubscriptionAdminClient client,
+    std::vector<std::string> const& argv) {
+  //! [delete-subscription]
```
Comment (alevenberg) [inline google/cloud/pubsub/samples/subscription_admin_samples.cc:83]:
```suggestion
```

File: google/cloud/pubsub/samples/subscription_admin_samples.cc [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    // Note that kNotFound is a possible result when the library retries.
+    if (status.code() == google::cloud::StatusCode::kNotFound) {
+      std::cout << "The subscription was not found\n";
+      return;
+    }
+    if (!status.ok()) throw std::runtime_error(status.message());
+
+    std::cout << "The subscription was successfully deleted\n";
+  }
+  //! [delete-subscription]
```
Comment (alevenberg) [inline google/cloud/pubsub/samples/subscription_admin_samples.cc:99]:
```suggestion
```

File: google/cloud/pubsub/samples/subscription_admin_samples.cc (hunk 1 above)
Comment (alevenberg) [reply google/cloud/pubsub/samples/subscription_admin_samples.cc]: Ack done

**Pull request merged**